### PR TITLE
Fix multi-asic behaviour for watermarkstat

### PR DESCRIPTION
**Pull request by bktsim-arista**
#### What I did
Added multi-asic support to watermarkstat, fixing watermark/persistent-watermark related commands. Previously, the following commands were not behaving correctly on multi-asic devices, as the '-n' namespace option was not available, and correct namespaces were not traversed on multi-asic devices.

* show buffer_pool watermark/persistent-watermark
* show headroom-pool watermark/persistent-watermark
* show priority-group persistent-watermark/watermark
* show queue persistent-watermark/watermark

This change fixes multi-asic behaviour of CLI commands that rely on watermarkstat as listed above.

This is a part of the set of changes being pushed for https://github.com/sonic-net/sonic-buildimage/issues/15148

#### How I did it
Added namespace option `-n` and used multi_asic library to implement multi_asic behaviour. Added relevant unit tests to ensure functionality.

#### How to verify it
Run unit tests, or the commands as listed above.

### CODE REVIEW

**Review by wenyiz2021 — DISMISSED** [review]
LGTM.
can you please provide proof that this change also works on single asic? I think there is UT that has run to prove

File: tests/multi_asic_watermarkstat_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+   def test_show_pg_shared_masic(self):
+      self.execute(['watermarkstat', '-t', 'pg_shared', '-n', 'asic1'],
+                   expected_result=show_pg_wm_shared_output_masic)
+
+   def test_show_headroom_pool_masic(self):
+      self.execute(['watermarkstat', '-t', 'headroom_pool', '-n', 'asic1'],
+                   expected_result=show_hdrm_pool_wm_output)
+
+   def test_show_invalid_asic_masic(self):
```
Comment (vmittal-msft) [inline tests/multi_asic_watermarkstat_test.py:38]: @bktsim-arista Few things -
1. If no namespace specified, Will it show for all namespaces? if we specify, will it show for only that namespace ?
2. Is this verified to be working fine on pizza box as well as chassis?

**Comment by vmittal-msft** [comment]
@bktsim-arista Can you please upate https://github.com/sonic-net/sonic-buildimage/issues/15148 on which PR is handling which command for easy tracking ?

**Comment by abdosi** [comment]
@alpeshspatel : please help review

**Comment by arista-hpandya** [comment]
The test failures are due to change in mock config DB, will push a fix addressing it soon.

**Comment by arista-hpandya** [comment]
Test snapshots from a mutli-asic T2 linecard:

### Show on the specified asic: watermarkstat -t q_shared_uni -n asic0
show queue watermark unicast asic0

### Show all asics when none specified:
show queue watermark unicast

### Perform clear on a single asic and verify no other asic was modified
show clear and show
Screenshot from 2024-09-16 15-36-09

**Review by wenyiz2021 — APPROVED** [review]
snapshot result :LGTM

**Comment by bingwang-ms** [comment]
@bktsim-arista Please raise another PR to 202405 to address cherry-pick conflict.

**Comment by arista-hpandya** [comment]
> @bktsim-arista Please raise another PR to 202405 to address cherry-pick conflict.

You can ping me or @kenneth-arista for such requests going ahead. Because the change touches pgdropstat, can we please get #3058 cherry-picked to 202405 first, this change has an overlap with the test files introduced in #3058

**Pull request merged**